test/controllers/auth/SelfAssessmentActionSpec.scala

```diff
@@ -56,32 +57,40 @@ class SelfAssessmentActionSpec
 
   val action = new SelfAssessmentActionImpl(citizenDetailsService, ninoAuthAction, appConfig)
 
-  class Harness(minAuthAction: AuthAction, selfAssessmentAction: SelfAssessmentAction) extends InjectedController {
+  class MyFakeAuthAction(utr: Option[SaUtr], uar: Option[Uar]) extends AuthAction with ControllerBaseSpec {
+
+    override val parser: BodyParser[AnyContent] = mcc.parsers.anyContent
+    override protected val executionContext: ExecutionContext = mcc.executionContext
+
+    override def invokeBlock[A](request: Request[A], block: AuthenticatedRequest[A] => Future[Result]): Future[Result] =
+      block(
+        AuthenticatedRequest(
+          "userId",
+          uar,
+          utr,
+          None,
+          None,
+          None,
+          None,
+          utr.isDefined,
+          uar.isDefined,
```

Comment:
Are these the right way round?

Reply:
yes, isSA comes before isAgentActive in the AuthenticatedRequest